crates/ibc/src/applications/transfer/relay/send_transfer.rs

```diff
@@ -1,33 +1,109 @@
-use crate::applications::transfer::context::TokenTransferContext;
+use crate::applications::transfer::context::{
+    TokenTransferExecutionContext, TokenTransferValidationContext,
+};
 use crate::applications::transfer::error::TokenTransferError;
 use crate::applications::transfer::events::TransferEvent;
 use crate::applications::transfer::msgs::transfer::MsgTransfer;
 use crate::applications::transfer::packet::PacketData;
 use crate::applications::transfer::{is_sender_chain_source, Coin, PrefixedCoin};
-use crate::core::ics04_channel::handler::send_packet::send_packet;
+use crate::core::ics04_channel::handler::send_packet::{send_packet_execute, send_packet_validate};
 use crate::core::ics04_channel::packet::Packet;
 use crate::core::ics24_host::path::{ChannelEndPath, SeqSendPath};
 use crate::events::ModuleEvent;
-use crate::handler::{HandlerOutput, HandlerOutputBuilder};
 use crate::prelude::*;
 
-/// This function handles the transfer sending logic.
-/// If this method returns an error, the runtime is expected to rollback all state modifications to
-/// the `Ctx` caused by all messages from the transaction that this `msg` is a part of.
```

Comment:
Did this no longer correspond with the changes?

Reply:
I added it back [2f1fce1](https://github.com/cosmos/ibc-rs/pull/468/commits/2f1fce19a5a14214682a09d5d19acf31470e4b48)

However this is true in general with `Validation/ExecutionContext`, so we should remember to document it more widely with #376.

Reply:
Agreed 👍🏻